src/Tools/Console/ConsoleRunner.php

```diff
@@ -25,7 +28,13 @@ class ConsoleRunner
      */
     public static function run(ConnectionProvider $connectionProvider, $commands = [])
     {
-        $cli = new Application('Doctrine Command Line Interface', Versions::getVersion('doctrine/dbal'));
+        $version = '';
+        if (class_exists(InstalledVersions::class)) {
```

Comment:
Adding a dependency on Composer 2 runtime API isn't a breaking change. If you're using Composer 1, you won't be able to upgrade and will keep using the version that works with Composer 1.

It's better to add a dependency and motivate those who are behind to upgrade rather than taking this technical debt on ourselves.

Reply:
> Adding a dependency on Composer 2 runtime API isn't a breaking change.

True. I was reluctant to add it because we use the runtime API for non-essential functionality of an optional feature. But I'm very much fine with adding it.